### ISSUE TITLE
Decoupling Android and Flutter app

### ISSUE THREAD

**Issue by anhappdev** [mlcommons/mobile_app_open#249]
The Android and Flutter app are currently depended on the same module `android/cpp`.
There are some issues (Flutter app related) which require changes in `android/cpp`. Since we are going to deprecate Android in the next version, I suggest we **copy** `android/cpp` into `flutter/cpp` so that the changes don't impact the Android app.

**Comment by anhappdev** [comment]
@relja128 and @d-uzlov, are you ok with this?

**Comment by d-uzlov** [comment]
I, OK with this.

**Comment by freedomtan** [comment]
copying `android/cpp` and to `flutter/cpp` and modifying it sounds like carrying technical debt to me :-)

**Comment by anhappdev** [comment]
The point is, we only need to maintain both Android and Flutter for a short time. After that, only the Flutter app is actively maintained and the Android app will be deprecated and archived.

**Comment by freedomtan** [comment]
Let's delay the decision to next meeting.

**Comment by anhappdev** [comment]
I summarize all the suggestions we have here for better overview:
1. **Copy** `android/cpp` into `flutter/cpp`, maintain both apps for a short time, then deprecate Android app.
2. Create a new branch to maintain the Android app, the Flutter app will be maintained in the main branch.
3. **Move** `android/cpp` to the root directory and versioning the API.

**Comment by d-uzlov** [comment]
>1. Copy android/cpp into flutter/cpp, maintain both apps for a short time, then deprecate Android app.

We would also need to make a copy of all backends that we want to use with the old Android app.

**Comment by anhappdev** [comment]
> We would also need to make a copy of all backends that we want to use with the old Android app.

Thanks, updated.

**Comment by freedomtan** [comment]
@freedomtan and @jwookiehong to bring this to policy meeting this week

**Comment by jwookiehong** [comment]
Policy meeting conclusion: copy android/cpp into flutter/cpp, and eventually deprecate android app.

Refactoring flutter/cpp is something we can consider doing at a later date.   
@freedomtan Does this look good?

**Comment by freedomtan** [comment]
Yup, @anhappdev and @d-uzlov let's make #250 happen

**Comment by anhappdev** [comment]
So I am going with solution number 2, which is
Create a new branch to maintain the Android app, the Flutter app will be maintained in the main branch.

This is similar to solution number 1 but we don't have code duplication in one branch.

**Comment by anhappdev** [comment]
There is now a new branch https://github.com/mlcommons/mobile_app_open/tree/android-v2 for maintaining Android app.
Flutter app will be further developed on the main branch.
1. With this PR https://github.com/mlcommons/mobile_app_open/pull/250, `android/cpp` is copied into `flutter/cpp` and all `mobile_back_*` are changed to point to `flutter/cpp`.
2. I will create a new PR next week to remove `android` folder and it's dependencies from the main branch. After that, we have 2 separated branches for Android and Flutter.

**Comment by d-uzlov** [comment]
We should also make `android-v2` branch protected to only allow changes via a pull request, and we will need to set up CI checks for it.

**Comment by d-uzlov** [comment]
I have set up branch protection for `android-v2` branch. It now has all the same rules as the master branch, except for the `dart` format check and checks that build the flutter app for ios and android.
I have also edited Google Cloud Build configuration to trigger build script for the old app for commits and PRs to the `android-v2` branch instead of `master`, and removed the requirement for corresponding check from rules for the `master` branch.
